presto-manager-common/src/main/java/com/teradata/prestomanager/common/NodeSet.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.common;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+public class NodeSet
+{
+    public static Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
+
+    public static void addAgent(URI uri, NodeRole nodeRole)
+    {
+        nodeSet.add(new Agent(uri, nodeRole));
+    }
+
+    public static void removeAgent(Integer id)
+    {
+        nodeSet.removeIf(agent -> agent.getId().equals(id));
+    }
+
+    public static Collection<URI> getAllNodes()
+    {
+        return nodeSet.stream().map(Agent::getUri).collect(Collectors.toList());
+    }
+
+    public static Collection<URI> getCoordinator()
+    {
+        return nodeSet.stream().filter(agent -> agent.getNodeRole() == NodeRole.COORDINATOR).map(Agent::getUri).collect(Collectors.toList());
+    }
+
+    public static Collection<URI> getWorkers()
+    {
+        return nodeSet.stream().filter(agent -> agent.getNodeRole() == NodeRole.WORKER).map(Agent::getUri).collect(Collectors.toList());
+    }
+
+    public enum NodeRole
+    {
+        WORKER, COORDINATOR
+    }
+
+    //TODO: Add method to generate ID
+    public static class Agent
+    {
+        private URI uri;
+        private static final AtomicInteger count = new AtomicInteger(0);
+        private int id;
```

Comment:
We discussed that a bit with Christina and Maciej in sync-up today.

We'll likely end up using a UUID eventually, but we're sticking with this at least until we have a  discovery service.